### ISSUE TITLE
Improve the prior interface for `make_epi_model_inference` 

### ISSUE THREAD

**Issue by seabbs** [CDCgov/Rt-without-renewal#45]
As the title says we should be able to improve the prior interface. The simplest improvement would be to generalise the argument name for the latent process priors to reflect that it is used across all submodules. An additional improvement would be to think more deeply about the coupling of priors to model submodules.

**Comment by SamuelBrand1** [comment]
> The simplest improvement would be to generalise the argument name for the latent process priors to reflect that it is used across all submodules.

Yes, atm I just use `merge` on the `NameTuple`s of each set of prior distributions.

> An additional improvement would be to think more deeply about the coupling of priors to model submodules.

I'm going to try and prototype this with the easy fix as back up. It _shouldn't_ be very hard. Basic idea is to wrap the function that makes a `Turing` model for each process with its priors in a struct. **If** you think thats cleaner than having a `NamedTuple` of priors lets do that.

The essential question is:

- Should all the priors "live" together in code as one object?
- Should the priors of a particular process "live" with the process definition as one object?

**Comment by seabbs** [comment]
> I'm going to try and prototype this with the easy fix as back up. It shouldn't be very hard. Basic idea is to wrap the function that makes a Turing model for each process with its priors in a struct. If you think thats cleaner than having a NamedTuple of priors lets do that.

Yes I think this is a nice idea. 

> Should the priors of a particular process "live" with the process definition as one object?

This seems cleanest to me and I think likely to have fewer failure states.
